packages/core/src/__tests__/caip10-link.test.ts

```diff
@@ -122,6 +122,23 @@ describe('Ceramic API', () => {
       expect(link.state).toMatchSnapshot()
     })
 
+    it('Create and link DID with legacy CAIP format', async () => {
+      const account = '0x0544DcF4fcE959C6C4F3b7530190cB5E1BD67Cb7@eip155:1'
+      const linkProof = { account, did: ceramic.did.id }
+      authProvider.createLink.mockReturnValueOnce(linkProof)
+
+      const link = await Caip10Link.fromAccount(ceramic, account)
+      await link.setDid(ceramic.did, authProvider, { anchor: false })
+
+      expect(link.did).toEqual(ceramic.did.id)
+      expect(link.state.log).toHaveLength(2)
+      expect(authProvider.createLink).toHaveBeenCalledTimes(1)
+      // toHaveBeenCalledTimes aggregates values over multiple tests
```

Comment:
I tried doing that but it caused like 5 other tests to fail - not exactly sure why since they weren't directly doing anything with validateLink but I think it might be because the `mockImplementation` got overridden

Reply:
Ah I can use `mockClear` instead of `mockReset` which won't override the implementation

Reply:
ah yes, my bad